components/server/src/workspace/workspace-service.ts

```diff
@@ -740,43 +739,28 @@ export class WorkspaceService {
         preference: WorkspaceRegion,
         clientCountryCode: string | undefined,
     ): Promise<WorkspaceRegion> {
-        const guessWorkspaceRegionEnabled = await getExperimentsClientForBackend().getValueAsync(
-            "guessWorkspaceRegion",
-            false,
-            {
-                user: { id: userId || "" },
-            },
-        );
-
-        const regionLogContext = {
-            requested_region: preference,
-            client_region_from_header: clientCountryCode,
-            experiment_enabled: false,
-            guessed_region: "",
-        };
-
         let targetRegion = preference;
         if (!isWorkspaceRegion(preference)) {
             targetRegion = "";
-        } else {
-            targetRegion = preference;
         }
 
-        if (guessWorkspaceRegionEnabled) {
-            regionLogContext.experiment_enabled = true;
-
-            if (!preference) {
-                // Attempt to identify the region based on LoadBalancer headers, if there was no explicit choice on the request.
-                // The Client region contains the two letter country code.
-                if (clientCountryCode) {
-                    targetRegion = RegionService.countryCodeToNearestWorkspaceRegion(clientCountryCode);
-                    regionLogContext.guessed_region = targetRegion;
-                }
+        let guessedRegion = "";
+        if (!preference) {
```

Comment:
Should this depend on `targetRegion` instead, since `preference` is already used above?
```suggestion
        if (!targetRegion) {
```

Reply:
It's true this has not been changed here, so feel free to just merge despite it.

Reply:
> It's true this has not been changed here, so feel free to just merge despite it.

:dart: And I wondered if it might make sense from the API perspective: only guess if there was not preference passed (explicitly).
Basically did not want to go into more detail here, as that would have involved writing unit tests, so going with what we had before.